src/core/router/info.cc

```diff
@@ -318,10 +319,23 @@ void RouterInfo::ParseRouterInfo(const std::string& router_info)
                 address.mtu = boost::lexical_cast<std::uint16_t>(value);
                 break;
               case Trait::Key:
-                // Our intro key as introducer
-                kovri::core::Base64ToByteStream(
-                    value.c_str(), value.size(), address.key, 32);
-                break;
+                {
+                  // Our intro key as introducer
+                  std::vector<std::uint8_t> key;
+                  try
+                    {
+                      key = core::Base64::Decode(value.c_str(), value.size());
+                    }
+                  catch (...)
+                    {
+                      m_Exception.Dispatch("RouterInfo: invalid intro key trait");
+                      is_valid_address = false;
+                    }
+
+                  //TODO(anonimal): let's try to avoid a memcpy
+                  std::memcpy(address.key, key.data(), sizeof(address.key));
```

Comment:
not easily searchable for a secure alternative. from (cppreference)[http://en.cppreference.com/w/cpp/string/byte/memcpy]:
- If either dest or src is a null pointer, the behavior is undefined, even if count is zero.
- If the objects are not TriviallyCopyable, the behavior of memcpy is not specified and may be undefined

For the first case, checking that buffers are non-null seems like an easy fix.

Maybe for the `TriviallyCopyable` case, could use:
```
try {
bool address_key_trivially_copyable = is_trivially_copyable<address.key>::value;
bool key_trivially_copyable ...

if !address_key_trivially_copyable || !key_trivially_copyable
    throw
}
catch (...)
```

Know this doesn't fully address the "secure memcpy" problem, but would it help make the current situation more secure?